src/wasm-interpreter.h

```diff
@@ -2006,6 +2006,17 @@ class ConstantExpressionRunner : public ExpressionRunner<SubType> {
     globalValues[name] = values;
   }
 
+  Flow visitDrop(Drop* curr) {
+    NOTE_ENTER("Drop");
+    Flow value = ExpressionRunner<SubType>::visit(curr->value);
+    if (!(flags & FlagValues::PRESERVE_SIDEEFFECTS)) {
+      value.clearIf(NONCONSTANT_FLOW);
```

Comment:
I don't remember this code well, but can't various things lead to `NONCONSTANT_FLOW`? I'm not sure they are all ok to ignore here, even if we ignore side effects.

Reply:
I think changing to a new error is better and can avoid this issue. So this behavior will be more robust and not be changed by some behavior in base class .